graylog2-server/src/main/java/org/graylog/plugins/views/search/jobs/SearchJobState.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.jobs;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import org.graylog.plugins.views.search.SearchJobIdentifier;
+import org.graylog2.database.MongoEntity;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.mongojack.Id;
+import org.mongojack.ObjectId;
+
+import java.util.List;
+import java.util.Map;
+
+@AutoValue
+@JsonDeserialize(builder = SearchJobState.Builder.class)
+public abstract class SearchJobState implements MongoEntity {
+
+    public static final String CREATED_AT_FIELD = "created_at";
+    public static final String UPDATED_AT_FIELD = "updated_at";
+    public static final String STATUS_FIELD = "status";
+    public static final String TYPE_FIELD = "type";
+    public static final String MESSAGES_FIELD = "messages";
+
+    @JsonUnwrapped
+    public abstract SearchJobIdentifier identifier();
+
+    @JsonProperty(STATUS_FIELD)
+    public abstract SearchJobStatus status();
+
+    @JsonProperty(TYPE_FIELD)
+    public abstract SearchJobType type();
+
+    @JsonProperty("error_message")
+    public abstract String errorMessage();
+
+    @JsonProperty("progress")
+    public abstract int progress();
+
+    @JsonProperty(MESSAGES_FIELD)
+    public abstract List<Map<String, Object>> messages(); //TODO: if we decide to store search engine results as well (not only Data Lake), we would need to keep a more general SearchType.Result object here
```

Comment:
I'm wondering if we want to make this reusable already before we have anything in MongoDB that needs a migration.

Reply:
Thx for a great suggestion!
Done! :)